### PR TITLE
Desktop: add nord + switch case

### PR DESCRIPTION
**Pull request by vzsky**
Desktop : Added Nord theme
by looking at the Nord's documentation and from previous trial (#1921)
provided brief documentation from Nord as of today in distinct file theme_nord.js
built with array for easy future edit 
change from confusing tonnes of if else to switch case

(since npm is broken on its side, the processes is untested)

### CODE REVIEW

File: ElectronClient/app/gui/NoteText.jsx [hunk 1]
```diff
@@ -49,6 +49,7 @@ require('brace/theme/solarized_light');
 require('brace/theme/solarized_dark');
 require('brace/theme/twilight');
 require('brace/theme/dracula');
+require('brace/theme/nord');
```
Comment (laurent22) [inline ElectronClient/app/gui/NoteText.jsx]: Is that a valid Brace theme? I'm not seeing it in node_modules

File: ReactNativeClient/lib/models/Setting.js [hunk 2]
```diff
@@ -886,6 +886,7 @@ Setting.THEME_DARK = 2;
 Setting.THEME_SOLARIZED_LIGHT = 3;
 Setting.THEME_SOLARIZED_DARK = 4;
 Setting.THEME_DRACULA = 5;
+Setting.THEME_NORD = 6;
```
Comment (laurent22) [inline ReactNativeClient/lib/models/Setting.js]: Please make sure the theme is only available on desktop (mobile and desktop don't share the same theme files).

File: ElectronClient/app/theme_nord.js [hunk 3]
```diff
@@ -0,0 +1,86 @@
+const nord = ['#2e3440', '#3b4252', '#434c5e', '#4c566a', '#d8dee9', '#e5e9f0', '#eceff4', '#8fbcbb', '#88c0d0', '#81a1c1', '#5e81ac', '#bf616a', '#d08770', '#ebcb8b', '#a3be8c', '#b48ead'];
```
Comment (laurent22) [inline ElectronClient/app/theme_nord.js]: It's a good idea to move the theme to a separate file, but please could you move it under app/gui/style/theme?

**Comment by laurent22** [comment]
Thanks for the PR, I've left a few comments. Also please could you provide a screenshot?

**Comment by vzsky** [comment]
I've no idea about requiring _brace/theme/nord_ .
I commented it out just to avoid errors.

Other things should be fixed and the screenshot is provided

**Comment by laurent22** [comment]
Thanks, how about the syntax highlighting on the editor?

I don't know if a Nord theme is available but maybe you can find one that closely matches there: https://ace.c9.io/build/kitchen-sink.html Then you'll need to use the name from here: https://github.com/thlorenz/brace/tree/master/theme

**Comment by vzsky** [comment]
Provided a screenshot.
All issues are solved.

Screen Shot 2562-10-05 at 22 00 39

**Comment by laurent22** [comment]
Thanks, that looks good but please can you undo the changes to package.json and package-lock.json (total of 3 files) as they are not related to this PR.

**Comment by tessus** [comment]
@laurent22 The OP has reverted the changes to the files you mentioned.

**Comment by laurent22** [comment]
There's still a mod change on these two files, and [this](https://github.com/laurent22/joplin/pull/1946#discussion_r331754365) needs to be resolved.

**Comment by tessus** [comment]
@laurent22 I removed the commented code.

Btw, something is off with the linter.

IMO a case construct should look like this:

```
switch(theme) {
	case Setting.THEME_DARK:
		output = Object.assign({}, output, darkStyle);
		break;
	case Setting.THEME_SOLARIZED_LIGHT:
[snip]
```

or the `break` on the same line as the line above, but what the case should be intended and it is not. The linter put the case right under the switch.

**Comment by laurent22** [comment]
Yes I agree the linting indeed seems to be off, although I've seen that style before. We could add a new linter rule so that it looks more like a classic switch case. The break on the same line is fine though.

About the changes, there's still the package files mod bits that have been changed but shouldn't. But really it would be better if @vzsky could get back to us and finish the pull request. There's not much left to do.

**Comment by tessus** [comment]
@vzsky are you planning to make the requested changes?

**Comment by tessus** [comment]
This looks rather abandoned.

**Pull request closed without merging**